lib/ansible/galaxy/collection/__init__.py

```diff
@@ -1304,6 +1304,7 @@ def _build_collection_tar(
         file_manifest,  # type: FilesManifestType
 ):  # type: (...) -> str
     """Build a tar.gz collection artifact from the manifest data."""
+    file_manifest["files"].sort(key=lambda x: x["name"])
```

Comment:
It's probably more robust to use `sort_keys=True` in the `json.dumps` calls below.  Probably makes sense to do it in both files and manifest.

Reply:
I don't think json.dumps will sort a value even though it's a list of dictionaries.

Reply:
Ah good point, might be worth doing both then?

Reply:
@sivel from my understanding is that if we sort_keys = True on what we have which is a list of dictionaries as mentioned by @s-hertel , its not going to sort the dictionaries inside the list, which is the reason why I did not adopt the approach of using sort_keys in the first place. But I am happy to also add sort_keys=True in the json.dumps aswell.

Reply:
Yes, please do